src/main/java/software/amazon/smithy/gradle/SmithyPlugin.java

```diff
@@ -84,12 +85,19 @@ public void apply(Project project) {
             registerSourceSets(project);
             addCliDependencies(project);
             buildJarTask.setAllowUnknownTraits(extension.getAllowUnknownTraits());
+            buildJarTask.setSmithyBuildConfigs(extension.getSmithyBuildConfigs());
+            buildJarTask.setProjection(extension.getProjection());
+            buildJarTask.getProjectionSourceTags().addAll(extension.getProjectionSourceTags());
             // We need to manually add these files as a dependency because we
             // dynamically inject the Smithy CLI JARs into the classpath.
             Configuration smithyCliFiles = project.getConfigurations().getByName("smithyCli");
             buildJarTask.getInputs().files(smithyCliFiles);
         });
 
+        project.getTasks().withType(SmithyBuild.class).configureEach(smithyBuild -> {
```

Comment:
I'd like to not lose any debug logging with this change. Could you add that back in here?

```suggestion
        project.getTasks().withType(SmithyBuild.class).configureEach(smithyBuild -> {
            smithyBuild.setSmithyBuildConfigs(extension.getSmithyBuildConfigs());
            project.getLogger().debug("Setting smithyBuildConfigs of {} to {} from SmithyExtension",
                getClass().getName(), extension.getSmithyBuildConfigs());
            
```
```

Reply:
If we add the logs back literally like this it'll give customers the wrong impression on what the final value the task is actually configured to.  That is, your suggested change will always report `Setting smithyBuildConfigs of {} to {} from SmithyExtension`  even if the task is explicitly configured to a different value on the specific task instance because this configuration action will always execute before other task configuration actions.

My understanding of logging within the gradle ecosystem is that tasks generally log their effective configuration during their action execution.  Looking at `SmithyUtils.executeCli`, we're already covering this by logging the literal classpath and arguments being passed.